src/dmd/access.d

```diff
@@ -406,11 +406,8 @@ bool checkAccess(Loc loc, Scope* sc, Expression e, Declaration d)
     }
     if (!e)
     {
-        if (d.prot().kind == Prot.Kind.private_ && d.getAccessModule() != sc._module || d.prot().kind == Prot.Kind.package_ && !hasPackageAccess(sc, d))
-        {
-            error(loc, "%s `%s` is not accessible from module `%s`", d.kind(), d.toPrettyChars(), sc._module.toChars());
-            return true;
-        }
+        // Access checks have been removed in favor of visibility checks.
+        // See https://github.com/dlang/DIPs/blob/master/DIPs/archive/DIP22.md
```

Comment:
Shouldn't this whole function be reworked as a result ?
Not necessarily now tho.

Reply:
Yes, I'm hoping I can delete the entire function when I'm done.

Reply:
I couldn't get rid of it.  It is called for some safety checks, specifically https://github.com/dlang/dmd/blob/master/test/fail_compilation/test18554.d